### PR TITLE
Convert `Toolbar` to a function component

### PR DESCRIPTION
**Pull request by dewanshDT**
Changes:
Rewrote the old class based `Toolbar` to a functional component, along with new test file
 
I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] has no test errors (`npm run test`)
* [x] is from a uniquely-named feature branch and is up to date with the  `develop` branch.

### CODE REVIEW

File: client/modules/IDE/components/Header/Toolbar.jsx [hunk 1]
```diff
@@ -1,253 +1,192 @@
...
-    this.handleKeyPress = this.handleKeyPress.bind(this);
-    this.handleProjectNameChange = this.handleProjectNameChange.bind(this);
-    this.handleProjectNameClick = this.handleProjectNameClick.bind(this);
-    this.handleProjectNameSave = this.handleProjectNameSave.bind(this);
+const Toolbar = (props) => {
+  const { isPlaying, infiniteLoop, preferencesIsVisible } = useSelector(
+    (state) => state.ide
+  );
+  const project = useSelector((state) => state.project);
+  const { autorefresh } = useSelector((state) => state.preferences);
```
Comment (lindapaiste) [inline client/modules/IDE/components/Header/Toolbar.jsx]: It's better to select  only `state.preferences.autorefresh` than to select the whole object and destructure it.  That way the toolbar doesn't need to re-render if other preferences change.

File: client/modules/IDE/components/Header/Toolbar.jsx [hunk 2]
```diff
@@ -1,253 +1,192 @@
...
-      });
-    } else {
-      this.props.setProjectName(newProjectName);
-      this.props.hideEditProjectName();
-      if (this.props.project.id) {
-        this.props.saveProject();
-      }
+  function handleProjectNameSave() {
+    const newName = nameInputValue;
+    if (newName.length > 0) {
```
Comment (lindapaiste) [inline client/modules/IDE/components/Header/Toolbar.jsx:61]: The handling of 0 length is a bit weird in the current version. If you click outside the input it reverts the text but keeps the input open.  It seems like yours does nothing which is probably fine.

File: client/modules/IDE/components/Header/Toolbar.unit.test.jsx [hunk 3]
```diff
@@ -1,73 +1,85 @@
...
-    const props = renderComponent({ currentUser: 'not-me' });
+    renderComponent({ user: { username: 'not-me', id: 'not-me' } });
     const sketchName = screen.getByLabelText('Edit sketch name');
 
     fireEvent.click(sketchName);
 
     expect(sketchName).toBeDisabled();
     await waitFor(() =>
-      expect(props.showEditProjectName).not.toHaveBeenCalled()
+      expect(screen.getByLabelText('New sketch name').disabled).toBe(true)
```
Comment (lindapaiste) [inline client/modules/IDE/components/Header/Toolbar.unit.test.jsx]: use `.toBeDisabled()`

File: client/modules/IDE/selectors/project.js [hunk 4]
```diff
@@ -2,6 +2,7 @@ import { createSelector } from 'reselect';
 
 export const selectProjectOwner = (state) => state.project.owner;
 export const selectProjectId = (state) => state.project.id;
+export const selectProjectName = (state) => state.project.name;
```
Comment (lindapaiste) [inline client/modules/IDE/selectors/project.js:5]: ❤️

File: client/modules/IDE/components/Header/Toolbar.jsx [hunk 5]
```diff
@@ -1,253 +1,192 @@
...
+        disabled={infiniteLoop}
+      >
+        <PlayIcon focusable="false" aria-hidden="true" />
+      </button>
+      <button
+        className={playButtonClass}
+        onClick={() => {
+          props.syncFileContent();
+          dispatch(startSketch());
+          console.log('play button pressed');
```
Comment (lindapaiste) [inline client/modules/IDE/components/Header/Toolbar.jsx]: Delete this before we merge.

File: client/modules/IDE/components/Header/Toolbar.jsx [hunk 6]
```diff
@@ -1,253 +1,192 @@
...
+  const projectNameInputRef = useRef();
+  const [nameInputValue, setNameInputValue] = useState(project.name);
 
-  handleKeyPress(event) {
+  function handleKeyPress(event) {
     if (event.key === 'Enter') {
-      this.props.hideEditProjectName();
-      this.projectNameInput.blur();
+      dispatch(hideEditProjectName());
+      projectNameInputRef.current.blur();
```
Comment (lindapaiste) [inline client/modules/IDE/components/Header/Toolbar.jsx]: Put a `?.` here.  Just in case the ref hasn't been set.

**Review by lindapaiste — CHANGES_REQUESTED** [review]
Looks mostly good.  I need to load up the branch and make sure that everything works.

File: client/modules/IDE/components/Header/Toolbar.jsx (hunk 2 above)
Comment (dewanshDT) [reply client/modules/IDE/components/Header/Toolbar.jsx:61]: let me know If you want me to change this

File: client/modules/IDE/components/Header/Toolbar.unit.test.jsx (hunk 3 above)
Comment (dewanshDT) [reply client/modules/IDE/components/Header/Toolbar.unit.test.jsx]: got it!

File: client/modules/IDE/selectors/project.js (hunk 4 above)
Comment (dewanshDT) [reply client/modules/IDE/selectors/project.js:5]: haha, at that time I didn't know how to cherry-pick

**Comment by dewanshDT** [comment]
Hey @lindapaiste! I just updated this with all the suggested changes, please check it out

**Comment by lindapaiste** [comment]
@dewanshDT I'm really not sure what our plan is with the toolbar.  Like what changes we intend to make and in what order. 
Should I merge this and then resolve conflicts to apply #2238 on top of this?

**Comment by dewanshDT** [comment]
We can still use your changes with the hook I created `useSketchActions`

**Comment by dewanshDT** [comment]
yes I think we should merge this and then resolve the conflicts to apply #2238, and then we can discuss the future changes than we want to make

**Review by lindapaiste — APPROVED** [review]

**Pull request merged**